File: src/main/java/com/example/lumaresort/entities/Reserva.java
package com.example.lumaresort.entities;

import java.util.Date;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Reserva {

    private Integer idReserva;
    private Date fechaInicio;
    private Date fechaFin;
    private Integer cantidadPersonas;
    private String estado;

    private Cliente cliente;
}

File: Frontend/src/app/Models/Reserva.ts
import { Habitacion } from './Habitacion';
import { Servicio } from './Servicio';
import { Usuario } from './Usuario';
import { Premio } from './Premio';

export interface Reserva {
  idReserva: number;
  fechaInicio: string | Date;
  fechaFin: string | Date;
  cantidadPersonas: number;
  estado: string;
  usuario: Usuario;
  habitacion: Habitacion;
  servicios: Servicio[];
  premios?: Premio[];  // Lista de premios asignados a la reserva
}
